plugins/outputs/mqtt/README.md

```diff
@@ -32,10 +32,14 @@ This plugin writes to a [MQTT Broker](http://http://mqtt.org/) acting as a mqtt
   # tls_key = "/etc/telegraf/key.pem"
   ## Use TLS but skip chain & host verification
   # insecure_skip_verify = false
-
+  
   ## When true, metrics will be sent in one MQTT message per flush.  Otherwise,
   ## metrics are written one metric per MQTT message.
   # batch = false
+  
+  ## When true, metric will have RETAIN flag set, making broker cache entries until someone
+  ## actually reads it
```

Comment:
I'm new to retained messages, but I read https://www.hivemq.com/blog/mqtt-essentials-part-8-retained-messages and this description does not seem to match.  The way I understood it is that the broker saves the last received message and immediately sends it to consumers on connection, not that it saves the messages until someone reads them (this would be controlled with the qos setting).

That said, I don't think we need to define what retained messages are, that's MQTT's docs job,  how about we just say:
> Send messages with the retained flag set.

Reply:
Agreed. 
Yes, I misworded the intent. `Retain` makes broker remember _last message of each topic_ and replay it to everybody upon connection - this way you won't ever lose **last** readout, at the cost of possible replay.

OTOH just sending messages with _QoS>0_ and having broker (_mosquitto_ in my case) configured with persistent storage won't make messages cached (that was my initial intent). Broker needs some additional push to do so, for example unexpired _QoS>0_ subscription (that is - client doesn't need to be connected right now) _or_ defined push/replication with _QoS>0_ to another broker (that is - remote broker need not be available right now). This behavior is _orthogonal_ to `retain` flag.